FabNESO.py

```diff
@@ -31,7 +34,21 @@ def neso(
         solver: Which NESO solver to use.
         conditions_file_name: Name of conditions XML file in configuration directory.
         mesh_file_name: Name of mesh XML in configuration directory.
+        **kwargs: Additional kwargs will be passed to ensemble_tools.edit_parameters to
+                  create a temporary conditions file with these overriden parameters
     """
+    # If there have been additional parameters provided, create a copy of the
```

Comment:
At the moment if there is an error raised after the temporary conditions directory is created but before the final `shutil.rmtree` call then we will end up with a `tmp_configs` directory remaining on disk which will cause an error next time we try to run with a parameter override unless we manually clean up. 

We can use [the ability of the `tempfiles.TemporaryDirectory` class to be used as a context manager](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) here to make sure that the temporary directory will always be cleaned up even when an exception is raised, and also ensure it always gets a unique name, for example something like

```Python
    config_path = Path(fab.find_config_file_path(config))
    with (
        TemporaryDirectory(prefix=f"{config}_", dir=config_path.parent)
        if kwargs != {} else nullcontext()
    ) as temporary_config_directory:
        if kwargs != {}: 
            # If there have been additional parameters provided, create a copy of the
            # conditions file and edit the input parameters
            temporary_config_path = Path(temporary_config_directory)
            shutil.copytree(config_path, temporary_config_path, dirs_exist_ok=True)
            edit_parameters(temporary_config_path / conditions_file_name, kwargs)
            config = temporary_config_path.name  
        fab.with_config(config)
        fab.execute(fab.put_configs, config)
        fab.job(
            {
                "script": "neso",
                "neso_solver": solver,
                "neso_conditions_file": conditions_file_name,
                "neso_mesh_file": mesh_file_name,
            }
        )
```

with 

```Python
from contextlib import nullcontext
from tempfile import TemporaryDirectory
```

added to the imports.